### DISCUSSION TITLE
RTK Query with redux-persist

### DISCUSSION THREAD

**Discussion by AlbinCederblad** [reduxjs/redux-toolkit#1172] [Q&A]
What happens when i use these together? Will it use the persisted store or will it fetch?

**Comment by phryneas** [comment]
I just really wouldn't. 
That restored data could be all kinds of stale and when a user hits F5 they usually expect data to be up-to-date, not a week old or something. Also, while the store slice is restored, information about subscriptions might be problematic (because the "subscribing components" never exist, they can also never unmount and thus get never cleaned from the store).

So, I'd blacklist the api slice. 

If you want that stuff to be cached, do it with cache headers in your api. The browser will do all the caching for you, but also allow the user to clear the cache or force a refetch with ctrl+shift+r - so the browser would just behave more than the user expects.

**Reply by AlbinCederblad** [reply]
Ok, Thanks!

I have another question, if you don't mind!

When i've fetched an object/array using Query, and i for example want to modify the result or add a property to the result. How do i go about doing that? It seems right now that the objects are frozen. :)

I'm trying to add a "favorited": true/false property to the result based on an existing state

**Reply by phryneas** [reply]
Generally, RTK-Query is really just meant to cache responses that reflect the state on the server - should you call the `.refresh()` method all your changes would be gone. Also, should multiple responses for multiple queries share the same data, they will not all be updated, but just the ones that you manually update.

That said, you can update cache values using the [`api.utils.updateQueryData`](https://redux-toolkit.js.org/rtk-query/api/created-api/cache-management-utils#updatequerydata) method.

Generally though, I would keep this additional data in a normal redux slice on the side, stored by id, and just retrieve it where you need it.

**Comment by hariDasu** [comment]
for an offline-first experience, how would you recommend persisting API data while user does not have network connectivity in a mobile app? considering persisting API data from rtk-query until app reconnects
